src/groupByKey.ts

```diff
@@ -2,6 +2,17 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const grouped: GroupsMap[T] = {};
+
+  for (const item of items) {
+    const value: string = item[key];
```

Comment:
The type of `value` should not be hardcoded as `string`. Instead, it should be the type of the property of `T` that `key` refers to. This could be achieved by using a type assertion or a generic type constraint to ensure that `key` is indeed a key of `T` and that `T[key]` is assignable to `value`.